### ISSUE TITLE
Seacas: Unable to link `z.lib` to tests on Windows

### ISSUE THREAD

**Issue by johnwparent** [LLNL/spack#43354]
In #43195 changes were added to the Seacas package link lines that are inherently incompatible with supported platforms.

```
        if "~shared" in self.spec:
            options.append(self.define(f"{project_name_base}_EXTRA_LINK_FLAGS", "z;dl"))
```
adds `z.lib` and `dl.lib` to the Seacas link lines. These are not valid libraries in Windows, nor does linking to non absolute paths to libraries fit the Windows Spack link idiom.

(Why are these lines neccesary to begin with? It seems to me like Seacas's CMake should be handling those libraries in the link interface and this is a bandaid over a larger issue)

### Steps to reproduce the issue

```console
[spack] > spack install seacas
```

### Error message

<details><summary>Error message</summary>

<pre>
  ←[0;91m>> 827    FAILED: packages/seacas/libraries/exodus/test/testrdd.exe←[0m
     828    cmd.exe /C "cd . && "C:\Program Files\Microsoft Visual Studio\2022\
            Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cm
            ake.exe" -E vs_link_exe --intdir=packages\seacas\libraries\exodus\t
            est\CMakeFiles\testrdd.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~
            1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.e
            xe --manifests  -- C:\PROGRA~1\MICROS~4\2022\COMMUN~1\VC\Tools\MSVC
            \1438~1.331\bin\Hostx64\x64\link.exe /nologo packages\seacas\librar
            ies\exodus\test\CMakeFiles\testrdd.dir\testrdd.c.obj  /out:packages
            \seacas\libraries\exodus\test\testrdd.exe /implib:packages\seacas\l
            ibraries\exodus\test\testrdd.lib /pdb:packages\seacas\libraries\exo
            dus\test\testrdd.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /sub
            system:console  packages\seacas\libraries\exodus\exodus.lib  C:\ci\
            windows-windows10.0.22621-skylake\msvc-19.38.33135\netcdf-c-4.9.2-d
            6cnljdjr2v4cqq6agvhq356rit5ezu5\lib\netcdf.lib  C:\ci\windows-windo
            ws10.0.22621-skylake\msvc-19.38.33135\hdf5-1.14.3-qizjzc3m3spy6l4sg
            4db6mhrikruj64u\lib\hdf5_hl.lib  C:\ci\windows-windows10.0.22621-sk
            ylake\msvc-19.38.33135\hdf5-1.14.3-qizjzc3m3spy6l4sg4db6mhrikruj64u
            \lib\hdf5.lib  z.lib  dl.lib  kernel32.lib user32.lib gdi32.lib win
            spool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib
            advapi32.lib && cd ."
     829    LINK: command "C:\PROGRA~1\MICROS~4\2022\COMMUN~1\VC\Tools\MSVC\143
            8~1.331\bin\Hostx64\x64\link.exe /nologo packages\seacas\libraries\
            exodus\test\CMakeFiles\testrdd.dir\testrdd.c.obj /out:packages\seac
            as\libraries\exodus\test\testrdd.exe /implib:packages\seacas\librar
            ies\exodus\test\testrdd.lib /pdb:packages\seacas\libraries\exodus\t
            est\testrdd.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsyste
            m:console packages\seacas\libraries\exodus\exodus.lib C:\ci\windows
            -windows10.0.22621-skylake\msvc-19.38.33135\netcdf-c-4.9.2-d6cnljdj
            r2v4cqq6agvhq356rit5ezu5\lib\netcdf.lib C:\ci\windows-windows10.0.2
            2621-skylake\msvc-19.38.33135\hdf5-1.14.3-qizjzc3m3spy6l4sg4db6mhri
            kruj64u\lib\hdf5_hl.lib C:\ci\windows-windows10.0.22621-skylake\msv
            c-19.38.33135\hdf5-1.14.3-qizjzc3m3spy6l4sg4db6mhrikruj64u\lib\hdf5
            .lib z.lib dl.lib kernel32.lib user32.lib gdi32.lib winspool.lib sh
            ell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib
             /MANIFEST:EMBED,ID=1" failed (exit code 1181) with the following o
            utput:
  ←[0;91m>> 830    LINK : fatal error LNK1181: cannot open input file 'z.lib
</pre></details>

### Information on your system

* **Spack:** 0.22.0.dev0 (93ffa9ba5da998bf6ce45548401cc376a8c00f41)
* **Python:** 3.11.3
* **Platform:** windows-windows10.0.22621-skylake
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by johnwparent** [comment]
This fix has already been made and was simple, but for general future reference, changes like these break Windows support because that's not how linking works with MSVC, references to other libraries should be made through proper CMake interfaces or at the very least, absolute paths.

> Note: No action needed on anybody's part, just wanted to document this so others have something to refer to in the future.

@scheibelp @alalazo

**Comment by alalazo** [comment]
@gsjaardema FYI

**Comment by alalazo** [comment]
Fixed in #43356

**Comment by alalazo** [comment]
This package is not listed among:
```console
$ spack list -t windows
7zip  abseil-cpp  bzip2  clingo  clingo-bootstrap  cmake  cpuinfo  curl  glm  hdf5  jsoncpp  msmpi  nasm  netlib-lapack  ninja  openssl  perl  python  re2c  ruby  scons  sqlite  tcl  wgl  win-sdk  win-wdk  winbison  wrf
==> 28 packages
```
Wasn't the `windows` tag supposed to be on packages that can build on windows?
